src/schema/v2/me/myCollectionInfo.ts

```diff
@@ -46,6 +70,67 @@ export const myCollectionInfoFields = {
     type: new GraphQLNonNull(GraphQLInt),
     resolve: ({ artists_count }) => artists_count,
   },
+  artistInsightsCount: {
+    type: artistInsightsCountType,
+    resolve: async (_root, _args, context) => {
+      const { collectionArtistsLoader, userID } = context
+
+      if (!collectionArtistsLoader) return
+
+      const { body: artists } = await collectionArtistsLoader("my-collection", {
+        size: MAX_ARTISTS,
+        user_id: userID,
+        all: true,
+      })
+
+      let soloShowCount = 0
+      let groupShowCount = 0
+      let collectedCount = 0
+      let reviewedCount = 0
+      let biennialCount = 0
+      let activeSecondaryMarketCount = 0
+
+      const countInsights = (
+        insight: ReturnType<typeof getArtistInsights>[0]
+      ) => {
+        switch (insight.kind) {
+          case "SOLO_SHOW":
+            soloShowCount += 1
+            break
+          case "GROUP_SHOW":
+            groupShowCount += 1
+            break
+          case "REVIEWED":
+            reviewedCount += 1
+            break
+          case "COLLECTED":
+            collectedCount += 1
+            break
+          case "BIENNIAL":
+            biennialCount += 1
+            break
+          case "ACTIVE_SECONDARY_MARKET":
+            activeSecondaryMarketCount += 1
+            break
+        }
+      }
+
+      artists.map((artist) => {
+        getArtistInsights(artist).map((insight) => {
+          if (!!insight.kind) countInsights(insight)
+        })
+      })
```

Comment:
**Non-blocker:** Why use `map` when you don't intend to use its return value? Feels anti-pattern to me. `forEach` or good old `for` loop is sufficient.